thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/util/CacheUtilTest.java

```diff
@@ -0,0 +1,104 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.apache.pinot.thirdeye.util;
+
+import com.couchbase.client.java.document.json.JsonObject;
+import org.apache.pinot.thirdeye.detection.cache.TimeSeriesDataPoint;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class CacheUtilTest {
+
+  private static final String BUCKET = "bucket";
+  private static final String DIMENSION_KEY = "dimensionKey";
+  private static final String METRIC_ID = "metricId";
+  private static final String START = "start";
+  private static final String END = "end";
+
+  private static String metricUrn = "thirdeye:metric:1";
+  private static String metricUrnHash = "624972944";
+  private final long timestamp = 1234567;
+  private final long metricId = 1;
+  private final String dataValue = "100.0";
+  private final String bucketName = "TestBucket";
+
+  private TimeSeriesDataPoint dataPoint;
+  private JsonObject jsonObject;
+
+
+  @BeforeMethod
+  public void beforeMethod() {
+    dataPoint = new TimeSeriesDataPoint(metricUrn, timestamp, metricId, dataValue);
+    jsonObject = JsonObject.create();
+  }
+
+  @AfterMethod
+  public void afterMethod() {
+    dataPoint = null;
+    jsonObject = JsonObject.empty();
+  }
+
+  @Test
+  public void testHashMetricUrn() {
+    Assert.assertEquals(metricUrnHash, CacheUtils.hashMetricUrn(metricUrn));
+  }
+
+  @Test
+  public void testBuildDocumentStructureShouldMapToJsonObject() {
+    JsonObject mappedDataPoint = CacheUtils.buildDocumentStructure(dataPoint);
+
+    Assert.assertEquals(mappedDataPoint.getLong("time").longValue(), 1234567);
+    Assert.assertEquals(mappedDataPoint.getLong("metricId").longValue(), 1);
+    Assert.assertEquals(mappedDataPoint.getString(dataPoint.getMetricUrnHash()), "100.0");
+  }
+
+  @Test
+  public void testBuildDocumentStructureShouldMapNullDataValueToZero() {
+    dataPoint.setDataValue(null);
+    JsonObject mappedDataPoint = CacheUtils.buildDocumentStructure(dataPoint);
+
+    Assert.assertEquals(mappedDataPoint.getString(dataPoint.getMetricUrnHash()), "0");
+  }
+
+  @Test
+  public void testBuildDocumentStructureShouldMapNullStringDataValueToZero() {
+    dataPoint.setDataValue("null");
+    JsonObject mappedDataPoint = CacheUtils.buildDocumentStructure(dataPoint);
+
+    Assert.assertEquals(mappedDataPoint.getString(dataPoint.getMetricUrnHash()), "0");
+  }
+
+  @Test
+  public void testBuildQuery() {
+    jsonObject.put(BUCKET, bucketName)
+        .put(METRIC_ID, metricId)
+        .put(DIMENSION_KEY, CacheUtils.hashMetricUrn(metricUrn))
+        .put(START, 100)
+        .put(END, 200);
+
+    String query = CacheUtils.buildQuery(jsonObject);
+    String expectedQuery = "SELECT time, `624972944` FROM `TestBucket` WHERE metricId = 1 AND `624972944` IS NOT MISSING AND time BETWEEN 100 AND 200 ORDER BY time ASC";
```

Comment:
This is pretty brittle - if the formatting of the query changes in any way your test will fail (even if the query itself is still valid). It may not be worth fixing this, but it would be worth noting at the top of the test that it is fairly brittle incase someone makes a change to the query in the future and is confused why the test is failing.

Ideally you would run a small test instance of the DB running that you could test the query against and then verify the results are as expected (less brittle). But that might be a pain to set up.

Reply:
I think we have some setup for this already (or at least I saw some logs that other tests are doing this). I'll look into it, thanks!